edgelet/kube-client/src/client.rs

```diff
@@ -1,5 +1,13 @@
 // Copyright (c) Microsoft. All rights reserved.
 
+// We are using a nightly clippy during CI which doesn't like code that maps
+// a result followed by an unwrap_or/unwrap_or_else. It wants us to use the
+// map_or_else combinator instead. This is a nighly only API however. So till
+// one of the following happens, we allow this lint:
+//  * map_or_else gets stabilized
+//  * our CI switches to stable clippy
```

Comment:
A small correction:

1. It's not a stable-vs-nightly clippy problem - clippy does not behave differently between stable and nightly.

1. The lint is not telling you to use `Result::map_or_else`, which is indeed unstable. It's telling you to use `.ok().map_or_else()` which is `Option::map_or_else`. That said, this lint has been historically broken because it suggests this even when the `unwrap_or_else` part actually uses the `Err`, which it could not if it used `.ok()`.

Rather than `#![allow]`ing it, I suggest replacing it with a `match` expr instead. (That is what I did for the rest of the codebase.)

Reply:
Good catch. I missed the `.ok()` bit in the lint message. Switched to using a match and removed the lint suppression.